errors/errors.go

```diff
@@ -33,8 +28,18 @@ func (a Location) Before(b Location) bool {
 }
 
 func Errorf(format string, a ...interface{}) *QueryError {
+	// similar to fmt.Errorf, Errorf will wrap the last argument if it is an instance of error
+	var err error
+	if n := len(a); n > 0 {
+		if v, ok := a[n-1].(error); ok {
+			err = v
+		}
+	}
+
 	return &QueryError{
-		Message: fmt.Sprintf(format, a...),
+		Err:        err,
+		Message:    fmt.Sprintf(format, a...),
+		Extensions: make(map[string]interface{}),
```

Comment:
Initialising Extension fields while returning a QueryError to avoid `assignment to entry in nil map` error